src/applications/Job.java

```diff
@@ -50,4 +50,8 @@ public int getId() {
         return id;
     }
 
+	void putTaskOnMachineQueue(MachineShopSimulator machineShopSimulator, int p) {
```

Comment:
Aren't you putting a _Job_ on the queue here? Feels like the name of the method might be:

```suggestion
	void putJobOnMachineQueue(MachineShopSimulator machineShopSimulator, int p) {
```

Reply:
I agree.